src/io_worker.cpp

```diff
@@ -135,38 +135,39 @@ void IOWorker::add_pool(const Host::ConstPtr& host, bool is_initial_connection)
   }
 }
 
-bool IOWorker::execute(RequestHandler* request_handler) {
-  return request_queue_.enqueue(request_handler);
-}
-
-void IOWorker::retry(RequestHandler* request_handler) {
-  Address address;
-  if (!request_handler->get_current_host_address(&address)) {
-    request_handler->on_error(CASS_ERROR_LIB_NO_HOSTS_AVAILABLE,
-                              "All hosts in current policy attempted "
-                              "and were either unavailable or failed");
-    return;
+bool IOWorker::execute(const RequestHandler::Ptr& request_handler) {
+  request_handler->inc_ref(); // Queue reference
+  if (!request_queue_.enqueue(request_handler.get())) {
+    request_handler->dec_ref();
```

Comment:
I do agree with the error proneness of manually inc/dec ref-counted object. I think we try to avoid this in most parts of the driver and I've attempted to make improvements in this release. If there are specific cases where you think it can be fixed let me know. There are a few cases where this might not be avoided:

1) Returning a reference to the external API. This is because the fact that it's a ref-counted type is lost.
2) In async code where the type is erased or it's added to a data structure that doesn't manage lifetime (e.g. an intrusively linked list). There's likely some improvement that could be made here in some places in `Connection`. 
3)  I've attempted to use `MCMPQueue<RequestHandler::Ptr>` instead of `MSMPQueue<RequestHandler*>`. The problem is the queue holds on to request memory until another request overwrites it or until the queue is destroyed. One solution is to proactively set the item to an empty object, `T()`, when dequeuing it, however, I am almost certain that would cause data races. The problem with keeping the memory alive is the size of requests are determine by the integrating application and could be quite large and there's an expectation that when a request is done that memory will be reclaimed.

The driver does not use exceptions  and limits the use of std library methods that throw exceptions.